bob/bob.py

```diff
@@ -1,2 +1,10 @@
 def hey(phrase):
-    pass
+    if phrase.isspace() or not phrase:
+        return 'Fine. Be that way!'
+    if phrase.strip().find('?')+1 == len(phrase.strip()) and not phrase.isupper():
+        return 'Sure.'
+    if phrase.strip().find('?')+1 == len(phrase.strip()) and phrase.isupper():
+        return 'Calm down, I know what Im doing!'
+    if phrase.strip().find('!') and phrase.isupper():
```

Comment:
@zarniak Though it's correct to use `find` but it's more "Pythonic" to do `if '!' in phrase`. `find` gives you back an index which you don't need here. Also, I'd assume that the implementation of `a in b` might be optimized better than a multi-purpose `find` function.